lib/Chat/Parser/SystemMessage.php

```diff
@@ -719,19 +722,39 @@ protected function getFileFromShare(?Participant $participant, string $shareId):
 			]);
 		}
 
+		$fileId = $node->getId();
+		$isPreviewAvailable = $this->previewManager->isAvailable($node);
+
 		$data = [
 			'type' => 'file',
-			'id' => (string) $node->getId(),
+			'id' => (string) $fileId,
 			'name' => $name,
 			'size' => $size,
 			'path' => $path,
 			'link' => $url,
 			'etag' => $node->getEtag(),
 			'permissions' => $node->getPermissions(),
 			'mimetype' => $node->getMimeType(),
-			'preview-available' => $this->previewManager->isAvailable($node) ? 'yes' : 'no',
+			'preview-available' => $isPreviewAvailable ? 'yes' : 'no',
 		];
 
+		// If a preview is available, check if we can get the dimensions of the file from the metadata API
+		if ($isPreviewAvailable) {
```

Comment:
To avoid unnecessary DB queries (as per other comment), let's check for the iOS client atm (unless @mahibi  also needs the info for Android)